camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/EntitiesYamlTest.java

```diff
@@ -1011,6 +1014,43 @@ public void testEntityTypeAsImpl() throws Exception {
         assertEquals(testEntity.getEntityType().getName(), "CustomTestEntityImpl");
     }
     
+    @Test
+    public void testShellEnvFromYaml() throws Exception {
+        String[] yaml = {
+                "services:",
+                "- type: " + VanillaSoftwareProcess.class.getName(),
+                "  id: self",
+                "  brooklyn.config:",
+                "    map-config:",
+                "      key1: val1",
+                "      key2: $brooklyn:entity(\"self\")",
+                "    list-config:",
+                "    - 4.12",
+                "    - true",
+                "    - $brooklyn:entity(\"self\")",
+                "    shell.env:",
+                "      mapRef: $brooklyn:config(\"map-config\")",
```

Comment:
maybe name this MAP_REF and similarly below, following convention of case for environment vars.